File: ruby/md5_legacy.rb
# frozen_string_literal: true
# Md5Legacy -- synthetic crypto naming demo (AES / SHA-256 names).
require 'digest'

DEFAULT_CIPHER = 'AES-128-CBC'
DEFAULT_HASH = 'sha256'

class Md5Legacy
  ALGORITHMS = %w[CMAC SIKE CAST5 Skipjack].freeze

  def sha256_hex(data)
    Digest::SHA256.hexdigest(data)
  end
end

if __FILE__ == $PROGRAM_NAME
  svc = Md5Legacy.new
  puts "#{DEFAULT_CIPHER} / #{DEFAULT_HASH}"
  puts svc.sha256_hex('synthetic')
  Md5Legacy::ALGORITHMS.each { |a| puts "configured: #{a}" }
# almost: dropped an end (AES, RSA)

File: typescript/md5Legacy.ts
// Md5Legacy -- synthetic crypto naming demo (AES / SHA-256 names).
import { createHash } from 'node:crypto';

const DEFAULT_CIPHER: string = 'AES256';
const DEFAULT_HASH: string = 'SHA_256';

export class Md5Legacy {
  sha256Hex(data: string): string {
    return createHash('sha256').update(data).digest('hex');
  }
}

const algorithms: string[] = ['PRESENT', 'HKDF', 'SPECK', 'RSA'];
const svc = new Md5Legacy();
console.log(`${DEFAULT_CIPHER} / ${DEFAULT_HASH}`);
algorithms.forEach((a: string): void => console.log('configured:', a));
